src/Security/Authentication/Core/src/AuthenticationHandler.cs

```diff
@@ -71,8 +71,14 @@ protected HttpResponse Response
     /// <summary>
     /// Gets the <see cref="ISystemClock"/>.
     /// </summary>
+    [Obsolete("ISystemClock is obsolete, use TimeProvider instead.")]
     protected ISystemClock Clock { get; }
 
+    /// <summary>
+    /// Gets the current time, primarily for unit testing.
+    /// </summary>
+    protected TimeProvider Time { get; }
```

Comment:
```suggestion
    protected TimeProvider TimeProvider { get; }
```

?

Reply:
Resolved offline, most of the auth handler properties are named like this.

Reply:
I think `Clock` and `Time` are quite different. I'd expect `Time` to be a `DateTimeOffset` or something.

We don't call the `AuthenticationHandler.UrlEncoder` property `AuthenticationHandler.Url` or call the `OptionsMonitor` property `Options`.

Since we didn't discuss this or the `SecurityStampValidator.Time` property in API review, can mark the issue as ready-for-review again and email the api review aliases? We could discuss `time` vs `timerProvider` for the parameter names too. Next time, I think we'll wait until there's a proper API proposal before we approving APIs. There's no need to rush this. It's too easy to miss things just looking at the PR and breaking change announcement.